lib/dat-manager.js

```diff
@@ -111,6 +111,25 @@ module.exports = class Manager extends EventEmitter {
     }, cb)
   }
   _activate (dat) {
+    Object.defineProperty(dat, 'progress', {
+      get: () => {
+        const stats = dat.hyperStats && dat.hyperStats.get()
+
+        let progress = 0
+        let total = 0
+
+        if (dat.archive) {
+          progress += dat.archive.metadata.blocks - dat.archive.metadata.blocksRemaining()
+          total += dat.archive.metadata.blocks
+        }
+        if (stats) {
+          progress += stats.blocksProgress
```

Comment:
pfrazee also had some content progress issues with similar calculation but may have been solved with this pr: https://github.com/mafintosh/hyperdrive/pull/123. 

The issue was that some clients don't want the historic blocks so they never "finish".